services/sqs/src/test/java/software/amazon/awssdk/services/sqs/SqsBatchManagerTest.java

```diff
@@ -0,0 +1,389 @@
+            + "        </ChangeMessageVisibilityBatchResultEntry>\n"
+            + "    </ChangeMessageVisibilityBatchResult>\n"
+            + "</ChangeMessageVisibilityBatchResponse>", id1, id2);
+
+        stubFor(any(anyUrl()).willReturn(aResponse().withStatus(200).withBody(responseBody)));
+
+        List<ChangeMessageVisibilityRequest> requests = new ArrayList<>();
+        requests.add(createChangeVisibilityRequest());
+        requests.add(createChangeVisibilityRequest());
+
+        List<CompletableFuture<ChangeMessageVisibilityResponse>> responses = new ArrayList<>();
+        long startTime = System.nanoTime();
+        for (ChangeMessageVisibilityRequest request : requests) {
+            responses.add(batchManager.changeMessageVisibility(request));
+        }
+        long endTime = System.nanoTime();
+        CompletableFuture.allOf(responses.toArray(new CompletableFuture[0])).join();
+
+        assertThat(Duration.ofNanos(endTime - startTime).toMillis()).isLessThan(DEFAULT_MAX_BATCH_OPEN + 100);
+    }
+
+    @Test
+    public void changeVisibilityBatchFunctionReturnsWithError_completeMessagesExceptionally() {
+        String responseBody = "<Error>\n"
+                              + "<Code>CustomError</Code>\n"
+                              + "<Message>Foo bar</Message>\n"
+                              + "<RequestId>656c76696e6727732072657175657374</RequestId>\n"
+                              + "<HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>\n"
+                              + "</Error>";
+
+        stubFor(any(anyUrl()).willReturn(aResponse().withStatus(400).withBody(responseBody)));
+
+        List<ChangeMessageVisibilityRequest> requests = new ArrayList<>();
+        requests.add(createChangeVisibilityRequest());
+        requests.add(createChangeVisibilityRequest());
+
+        List<CompletableFuture<ChangeMessageVisibilityResponse>> responses = new ArrayList<>();
+        for (ChangeMessageVisibilityRequest request : requests) {
+            responses.add(batchManager.changeMessageVisibility(request));
+        }
+
+        CompletableFuture<ChangeMessageVisibilityResponse> response1 = responses.get(0);
+        CompletableFuture<ChangeMessageVisibilityResponse> response2 = responses.get(1);
+        assertThatThrownBy(response1::join).isInstanceOf(CompletionException.class).hasMessageContaining("400");
+        assertThatThrownBy(response2::join).isInstanceOf(CompletionException.class).hasMessageContaining("400");
+    }
+
+    @Mock
+    private BatchManager<SendMessageRequest, SendMessageResponse, SendMessageBatchResponse> mockSendMessageBatchManager;
+
+    @Mock
+    private BatchManager<DeleteMessageRequest, DeleteMessageResponse, DeleteMessageBatchResponse> mockDeleteMessageBatchManager;
+
+    @Mock
+    private BatchManager<ChangeMessageVisibilityRequest, ChangeMessageVisibilityResponse,
+        ChangeMessageVisibilityBatchResponse> mockChangeVisibilityBatchManager;
```

Comment:
nit, can we move the member variables to the top?

Reply:
Moved to the top